tests/main/interfaces-audio-playback-record/task.yaml

```diff
@@ -35,19 +35,21 @@ prepare: |
     session-tool -u test --prepare
     echo "session-tool -u test --restore" >>defer.sh
 
+    # Ensure that the socket is active but do not check the service.  For some
+    # reason, pulseaudio quits as soon as it is not needed and this means that
+    # it works when we use paplay but shuts down relatively quickly after
+    # playback. We use to run "pulseaudio --check" but this only worked when
```

Comment:
I don't think this is just "for some reason", it seems to be a documented behavior of pulseaudio (from 20.04 man page):

```
       --exit-idle-time=SECS
              Terminate the daemon after the last client quit and this time  in  seconds  passed.
              Use a negative value to disable this feature. Defaults to 20.

              When  PulseAudio  runs  in  the per-user mode and detects a login session, then any
              positive value will be reset to 0 so that PulseAudio will terminate immediately  on
              logout. A positive value therefore has effect only in environments where there's no
              support for login session tracking. A negative value can still be used  to  disable
              any automatic exit.

              When PulseAudio runs in the system mode, automatic exit is always disabled, so this
              option does nothing.
```

So from that, it seems that the reason PulseAudio exits like it does is that it detects it's running in a user session now, and automatically idles itself during the session when not in use, relying on activation within the session to start it, and logout from the session to fully stop it.

Reply:
This explains it.

The key is the login session. We do get a login session inside `session-tool -u test ...` but not after `--prepare` (session services do not have a logind session).

Reply:
I'll adjust the comment.

Reply:
Done